subworkflows/local/profiling.nf

```diff
@@ -371,13 +372,27 @@ workflow PROFILING {
                                     if (it[0].is_fasta) log.warn "[nf-core/taxprofiler] mOTUs currently does not accept FASTA files as input. Skipping mOTUs for sample ${it[0].id}."
                                     !it[0].is_fasta
                                 }
-                                .multiMap {
-                                    it ->
-                                        reads: [it[0] + it[2], it[1]]
-                                        db: it[3]
+                                .branch {
+                                    longread: it[0].instrument_platform == 'OXFORD_NANOPORE'
+                                    shortread: it[0].instrument_platform != 'OXFORD_NANOPORE'
                                 }
+        ch_input_for_motus_longread = ch_input_for_motus.longread
+                                        .multiMap {
+                                            it ->
+                                                reads:[it[0] + it[2], it[1]]
+                                                db: it[3]
+                                        }
+        ch_input_for_motus_shortread = ch_input_for_motus.shortread
+                                        .multiMap {
+                                            it ->
+                                                reads:[it[0] + it[2], it[1]]
+                                                db: it[3]
+                                        }
+        MOTUS_PREPLONG ( ch_input_for_motus_longread.reads, ch_input_for_motus_longread.db )
+
+        MOTUS_PROFILE ( MOTUS_PREPLONG.out.out.mix(ch_input_for_motus_shortread.reads), ch_input_for_motus_longread.db.mix(ch_input_for_motus_shortread.db) )
```

Comment:
You need to mix the two into one input channel beforehand, and use a multiMap to make sure the reads and the db channels are insync

Reply:
Yes @sofstam !